verification/verifier_helpers.go

```diff
@@ -298,3 +316,51 @@ func (v *Verifier) executePlugin(ctx context.Context, trustPolicy *TrustPolicy,
 
 	return response, nil
 }
+
+// extractCriticalStringExtendedAttribute extracts a critical string Extended attribute from a signer.
+func extractCriticalStringExtendedAttribute(signerInfo *signature.SignerInfo, key string) (string, error) {
+	attr, err := signerInfo.ExtendedAttribute(key)
+	// not exist
+	if err != nil {
+		return "", errExtendedAttributeNotExist
+	}
+	// not critical
+	if !attr.Critical {
+		return "", fmt.Errorf("%v is not a critical Extended attribute", key)
+	}
+	// not string
+	val, ok := attr.Value.(string)
+	if !ok {
+		return "", fmt.Errorf("%v from Extended attribute is not a string", key)
+	}
+	return val, nil
+}
+
+// getVerificationPlugin get plugin name from the Extended attributes.
+func getVerificationPlugin(signerInfo *signature.SignerInfo) (string, error) {
+	name, err := extractCriticalStringExtendedAttribute(signerInfo, VerificationPlugin)
+	if err != nil {
+		return "", err
+	}
+	// not an empty string
+	if strings.TrimSpace(name) == "" {
+		return "", fmt.Errorf("%v from Extended attribute is an empty string", VerificationPlugin)
+	}
+	return name, nil
+}
+
+// getVerificationPlugin get plugin version from the Extended attributes.
+func getVerificationPluginMinVersion(signerInfo *signature.SignerInfo) (string, error) {
+	version, err := extractCriticalStringExtendedAttribute(signerInfo, VerificationPluginMinVersion)
+	if err != nil {
+		return "", err
+	}
+	// empty version
+	if strings.TrimSpace(version) == "" {
+		return "", fmt.Errorf("%v from Extended attribute is an empty string", VerificationPluginMinVersion)
+	}
+	if !semVerRegEx.MatchString(version) {
+		return "", fmt.Errorf("%v from Extended attribute is not a is not valid SemVer", VerificationPluginMinVersion)
```

Comment:
typo: `from Extended attribute is not a valid SemVer`

Reply:
Updated